tests/test_grid_sampling.py

```diff
@@ -99,6 +99,25 @@ def test_grid_sample_eval(grid, xdim=60, ydim=60):
 
 
 @pytest.mark.parametrize('mode', ['scipy', 'jit'])
+def test_nearest_neighbour_interpolation(mode, k_sample_p, xdim=2, ydim=2, npart=81):
+    lon = np.linspace(0., 1., xdim, dtype=np.float32)
+    lat = np.linspace(0., 1., ydim, dtype=np.float32)
+    U = np.zeros((xdim, ydim), dtype=np.float32)
+    V = np.zeros((xdim, ydim), dtype=np.float32)
+    P = np.zeros((xdim, ydim), dtype=np.float32)
+    P[0, 0] = 1.
+    grid = Grid.from_data(U, lon, lat, V, lon, lat, mesh='flat',
+                          field_data={'P': np.asarray(P, dtype=np.float32)})
+    grid.P.interp_method = 'nearest'
+    xv, yv = np.meshgrid(np.linspace(0.1, 0.9, np.sqrt(npart)), np.linspace(0.1, 0.9, np.sqrt(npart)))
+    pset = grid.ParticleSet(size=np.size(xv), pclass=pclass(mode),
+                            lon=xv.flatten(), lat=yv.flatten())
+    pset.execute(k_sample_p, endtime=1, dt=1)
+    assert np.allclose(np.array([p.p for p in pset if p.lon < 0.5 and p.lat < 0.5]), 1.0, rtol=1e-5)
```

Comment:
Does this still work with a differently sized grid, eg. 3x3? If not, we might want to hard-code the 2x2 grid and remove `xdim=2` and `ydim=2` from the function args.